OpenRA.Game/Traits/Util.cs

```diff
@@ -146,5 +146,153 @@ public static int ApplyPercentageModifiers(int number, IEnumerable<int> percenta
 
 			return (int)a;
 		}
+
+		// Algorithm obtained from ftp://ftp.isc.org/pub/usenet/comp.sources.unix/volume26/line3d
+		public static IEnumerable<CPos> Raycast(CPos source, CPos target)
```

Comment:
Rotate the WPos coordinate system by 45 degrees so that your WPos axes are aligned with the grid, and do the same things you do in the square grid.